modules/binding-infra-node-workloads-using-taints-tolerations.adoc

```diff
@@ -0,0 +1,65 @@
+// Module included in the following assemblies:
+//
+// * machine_management/creating-infrastructure-machinesets.adoc
+
+[id="binding-infra-node-workloads-using-taints-tolerations_{context}"]
+= Binding infrastructure node workloads using taints and tolerations
+
+If you have an infra node that has the `infra` and `worker` roles assigned, you
+must configure the node so that user workloads are not assigned to it.
+
+.Prerequisites
+
+* Configure additional MachineSets in your {product-title} cluster.
+
+.Procedure
+
+. Use the following command to add a taint to the infra node to prevent
+scheduling user workloads on it:
++
+[source,terminal]
+----
+$ oc adm taint nodes <node-name> <key>:<effect>
+----
++
+For example:
++
+[source,terminal]
+----
+$ oc adm taint nodes node1 node-role.kubernetes.io/infra:NoSchedule
+----
++
+This example places a taint on `node1` that has key
+`node-role.kubernetes.io/infra` and taint effect `NoSchedule`. Nodes with the
+`NoSchedule` effect schedule only Pods that tolerate the taint, but allow existing
+Pods to remain scheduled on the node.
++
+[NOTE]
+====
+If a descheduler is used, Pods violating node taints could be evicted from the
+cluster.
+====
+
+. Add tolerations for the Pod configurations you want to schedule on the infra
+node, like router, registry, and monitoring workloads. Add the following code to
+the Pod specification:
++
+[source, yaml]
+----
+tolerations: 
+  - effect: NoSchedule <1>
+    key: node-role.kubernetes.io/infra <2>
+    operator: Exists <3>
+----
+<1> Specify the effect that you added to the node.
+<2> Specify the key that you added to the node.
+<3> Specify the `Exists` Operator to require a taint with the key `node-role.kubernetes.io/infra` to be present on the node. 
+
+[NOTE]
+====
+It is not possible to move Pods for an Operator installed via OLM to an infra
+node by label.
+====
```

Comment:
@jewzaam This note caused some confusion in the BZ. I assume your suggestion to schedule infra resources using `nodeSelector` is the answer for working around this Operator restriction? Is that correct? I can add some additional text here to clarify.

Reply:
@codyhoag the operator restriction is about the inability to influence the deployment created by OLM.  It's baked into the CSV shipped with the operator.  If the operator developer wants to support deployment of the operator pods to infra nodes need to look at building it into the CSV with `preferredDuringSchedulingIgnoredDuringExecution` so it can schedule to `worker` if `infra` are not available.